ibis/pandas/core.py

```diff
@@ -184,26 +208,33 @@ def execute_with_scope(expr, scope, aggcontext=None, clients=None, **kwargs):
         post_execute_=functools.partial(
             post_execute,
             scope=scope,
+            timecontext=timecontext,
             aggcontext=aggcontext,
             clients=clients,
             **kwargs,
         ),
         **kwargs,
     )[op]
-
     return result
 
 
 @trace
 def execute_until_in_scope(
-    expr, scope, aggcontext=None, clients=None, post_execute_=None, **kwargs
+    expr,
+    scope,
+    timecontext=None,
+    aggcontext=None,
+    clients=None,
+    post_execute_=None,
+    **kwargs,
 ):
     """Execute until our op is in `scope`.
 
     Parameters
     ----------
     expr : ibis.expr.types.Expr
     scope : Mapping
+    timecontext : TimeContext
```

Comment:
Optional[TimeContext]

Reply:
Just realized that in definition TimeContext is already Optional. I have made all timecontext typings match with each other.